XenModel/Actions/ParallelAction.cs

```diff
@@ -180,7 +183,7 @@ private void action_Completed(ActionBase sender)
             sender.Completed -= action_Completed;
             lock (_lock)
             {
-                _completedActionsCount++;
+                _completedActionsCount = _completedActionsCount + 1;

```

Comment:
I think this is overzealous flagging on behalf of VS, possibly because it does not detect that the assignement is within a lock. The ++ incerement is indeed a compound operation because first it reads the value, then it adds one, then writes it, but since it is happening within the lock, I don't see what can interrupt it. `_completedActionsCount = _completedActionsCount + 1;` is the same thing, it only differs syntactically. The real correction would be to use Interlocked.Increment, but as I said, I don't see why it is needed since we have locked and the line cannot be accessed by another thread.

Reply:
100, completely agree. I just made the change to remove the warning, since it's not actually changing the logic in any way.

Either way, it's only visual. Either we have slightly more verbose code, or we get a warning in the editor. I prefer having the former and getting rid of the latter, but it's just preference ofc.